src/figuresClasses.ts

```diff
@@ -1,11 +1,106 @@
-export interface Figure {}
+/* eslint-disable @typescript-eslint/lines-between-class-members */
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+  private sideA: number;
+  private sideB: number;
+  private sideC: number;
+
+  constructor(
+    public color: Color,
+    sideA: number,
+    sideB: number,
+    sideC: number,
+  ) {
+    if (sideA <= 0 || sideB <= 0 || sideC <= 0) {
+      throw new Error('your error message');
+    }
+
+    if (
+      sideA + sideB <= sideC ||
+      sideA + sideC <= sideB ||
+      sideC + sideB <= sideA
+    ) {
+      throw new Error('your error message');
+    }
+
+    this.color = color;
+    this.sideA = sideA;
+    this.sideB = sideB;
+    this.sideC = sideC;
```

Comment:
If you declare in a constructor as I mentioned typescript will do it for you, so these are redundant here

Fix in all places